api/src/auth/RoomAuth.ts

```diff
@@ -42,3 +56,23 @@ export const verifyRoomToken = (
         return false;
     }
 };
+
+type RoomActions = RoomAction['action'];
+
+export const hasPermission = (
+    action: RoomActions,
+    payload: RoomTokenPayload,
+) => {
+    switch (action) {
+        case 'mark':
+            return !payload.isSpectating;
+        case 'unmark':
+            return !payload.isSpectating;
+        case 'newCard':
+            return payload.isMonitor;
+        case 'changeColor':
+            return !payload.isSpectating;
+        default:
+            return true;
+    }
+};
```

Comment:
This feels overly simplified. This is probably good for now, but should we build upon this in the feature, having a list of permissions and which permission you need for which action might be more helpful.
Ultimately your choice, but since this is supposed to be futureproof I feel like this is too simple

Reply:
I'm not sure I agree with listing permissions, since it feels like it may be pretty common to have "negative permissions" (where a rank takes away permissions), which we already see with spectator mode. We could refactor permissions and make it so that it is purely additive, but that feel incongruent with the fact that the default mode is base player permissions. If someone feels really strongly about it I would be open to tweaking this though.